lib/web_console/middleware.rb

```diff
@@ -28,6 +28,12 @@ def call(env)
       request = create_regular_or_whiny_request(env)
       return @app.call(env) unless request.from_whitelited_ip?
 
+      if request.xhr?
+        unless request.acceptable?
```

Comment:
I've got a feeling that the unless-statement works even for web apps. I will dig into it.